op-node/rollup/derive/span_channel_out.go

```diff
@@ -166,20 +194,24 @@ func (co *SpanChannelOut) AddSingularBatch(batch *SingularBatch, seqNum uint64)
 	if err = co.compress(); err != nil {
 		return err
 	}
-	co.lastCompressedRLPSize = co.activeRLP().Len()
 
 	// if the channel is now full, either return the compressed data, or the compressed previous data
 	if err := co.FullErr(); err != nil {
-		// if there is only one batch in the channel, it *must* be returned
-		if len(co.spanBatch.Batches) == 1 {
+		// if it's the first batch of the channel, it *must* fit in
```

Comment:
```suggestion
		// if it's the first batch of the channel *and* that batch only contains one block, it *must* fit in
```

technically you could find yourself with only one batch and have multiple blocks inside it and you would would in that case want to roll back the last block of the batch. The code logic is correct, just the comment indicates that batch-count is all that matters.

Reply:
"batch" here refers to the _singular_ batch, stored in variable `batch`, which corresponds 1:1 to a single block. Will clarify by adding "singular".